File: src/main/java/employeetech/com/employee/manager/model/Department.java
package employeetech.com.employee.manager.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "departments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Department implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @OneToMany(mappedBy = "department", cascade = CascadeType.ALL)
    private List<Employee> employees;
}

File: src/main/kotlin/employeetech/com/employee/manager/model/Department.kt
package employeetech.com.employee.manager.model

import jakarta.persistence.*
import java.io.Serializable

@Entity
@Table(name = "departments")
data class Department(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    val id: Long? = null,

    @Column(nullable = false, unique = true)
    val name: String? = null,

    @OneToMany(mappedBy = "department", cascade = [CascadeType.ALL])
    val employees: List<Employee>? = null
) : Serializable
